internal/app/splinter/model/interface.go

```diff
@@ -0,0 +1,58 @@
+package model
+
+type UserInterface interface {
+	Register(request RegisterRequest) (*AuthenticateResponse, error)
+	Authenticate(request AuthenticateRequest) (*AuthenticateResponse, error)
+
+	GetUserById(token Token, id string) (*User, error)
```

Comment:
Why do you need to pass around a token? Once you know the actual user id, it should be sufficient for any usecases. Tokens come and go, especially JWTs. They are quite ephemeral..